### DISCUSSION TITLE
Can you have a keymap to toggle float width?

### DISCUSSION THREAD

**Discussion by gonstoll** [ibhagwan/fzf-lua#1548] [Q&A] [answered]
I like my fzf float to be pretty slim, but sometimes I jump into codebases with large filenames and nested folder structures. Would be nice to have a keymap where you can toggle the `winopts.width` global option directly without having to specifically pass it to pickers.

**Comment by ibhagwan** [comment]
> I like my fzf float to be pretty slim, but sometimes I jump into codebases with large filenames and nested folder structures. Would be nice to have a keymap where you can toggle the `winopts.width` global option directly without having to specifically pass it to pickers.

Have you tried full screen (default F2), or rotate to horizontal/vertical (F5/F6)?

**Comment by ibhagwan** [comment]
You can also use toggle_behavior extend to have the main window slim and the preview “extending” it, see https://github.com/ibhagwan/fzf-lua/issues/721#issuecomment-1499456577

**Comment by gonstoll** [comment]
@ibhagwan thanks for the quick response! None of the F keys seems to work actually 🤔😭 but to illustrate a bit what I mean, this is what I mean. See how basically all file paths are truncated with ellipsis?

It'd be fantastic to have a keymap that automatically toggles full screen of the fzf float.

![Screenshot 2024-12-05 at 17 31 39](https://github.com/user-attachments/assets/3dd54c3c-593c-4035-8b05-fb9afa82ebad)

**Reply by ibhagwan** [reply]
> None of the F keys seems to work actually

The F keys don’t work is most likely unrelated to fzf-lua, your term is sending different keys.

> It'd be fantastic to have a keymap that automatically toggles full screen of the fzf float.

There are keys to toggle full screen and also rotation (horizontal to vertical and vice versa), redefine `keymap.builtin` and use `ctrl` keys to test until you solve why your F keys aren’t working:
https://github.com/ibhagwan/fzf-lua/blob/d9db3b684384bcfd1393914b76f89f0205f183a3/README.md?plain=1#L622-L661

You can quickly test it with `ctrl-f`:
```lua
:lua require("fzf-lua").files({keymap={builtin={["<c-f>"]="toggle-fullscreen"}}})
```

Also try:
```lua
:lua require("fzf-lua").files({keymap={builtin={["<c-f>"]="toggle-prreview-ccw"}}})
```

**Reply by ibhagwan** [reply]
Btw, to see what your neovim sees when you use the F keys, go into insert mode press `ctrl-v` and then an F key, you should expect to see `<F1>`, if something else shows up you’ve found your issue.

**Reply by gonstoll** [reply, marked as answer]
Aaah, this was my bad. My term was correctly sending the F keys, I just forgot to add `true` on my custom keymaps, so I was not extending the default ones. It works now, thank you so much for the help @ibhagwan! Fzf-lua is awesome ❤️

**Reply by ibhagwan** [reply]
> Aaah, this was my bad. My term was correctly sending the F keys, I just forgot to add `true` on my custom keymaps, so I was not extending the default ones. It works now, thank you so much for the help @ibhagwan! Fzf-lua is awesome ❤️

Ty for the update and kind words, cheers :-)
